test/unit/sessions.test.js

```diff
@@ -181,9 +159,147 @@ describe('Sessions - unit/core', function () {
+        expect(
+          () => new ClientSession(client, sessionPool, { causalConsistency: true, snapshot: true })
+        ).to.throw('Properties "causalConsistency" and "snapshot" are mutually exclusive');
+      });
+
+      it('should default to `null` for `clusterTime`', function () {
+        const client = new Topology('localhost:27017', {});
+        sessionPool = client.s.sessionPool;
+        session = new ClientSession(client, sessionPool);
+        expect(session.clusterTime).to.not.exist;
+      });
+
+      it('should set the internal clusterTime to `initialClusterTime` if provided', function () {
+        const clusterTime = genClusterTime(Date.now());
+        const client = new Topology('localhost:27017');
+        sessionPool = client.s.sessionPool;
+        session = new ClientSession(client, sessionPool, { initialClusterTime: clusterTime });
+        expect(session.clusterTime).to.eql(clusterTime);
+      });
+      it('should acquire a serverSession in the constructor if the session is explicit', () => {
+        const session = new ClientSession(topology, serverSessionPool, { explicit: true });
+        const serverSessionSymbol = getSymbolFrom(session, 'serverSession');
+        expect(session).to.have.property(serverSessionSymbol).that.is.an.instanceOf(ServerSession);
+      });
+
+      it('should leave serverSession null if the session is implicit', () => {
+        // implicit via false (this should not be allowed...)
+        let session = new ClientSession(topology, serverSessionPool, { explicit: false });
+        const serverSessionSymbol = getSymbolFrom(session, 'serverSession');
+        expect(session).to.have.property(serverSessionSymbol, undefined);
+        // implicit via omission
+        session = new ClientSession(topology, serverSessionPool, {});
+        expect(session).to.have.property(serverSessionSymbol, undefined);
+      });
+
+      it('should start the txnNumberIncrement at zero', () => {
+        const session = new ClientSession(topology, serverSessionPool);
+        const txnNumberIncrementSymbol = getSymbolFrom(session, 'txnNumberIncrement');
+        expect(session).to.have.property(txnNumberIncrementSymbol, 0);
+      });
+    });
+
+    describe('get serverSession()', () => {
+      it('should return whatever is defined for serverSession symbol if clientSession.hadEnded is true', () => {
+        const session = new ClientSession(topology, serverSessionPool, { explicit: false });
+        const serverSessionSymbol = getSymbolFrom(session, 'serverSession');
+        expect(session).to.have.property(serverSessionSymbol, undefined);
+        session.hasEnded = true;
+        expect(session.serverSession).to.be.undefined;
+        session[serverSessionSymbol] = 'wacky crazy value';
+        expect(session.serverSession).to.be.equal('wacky crazy value');
+      });
+
+      it('should acquire a serverSession if clientSession.hadEnded is false', () => {
```

Comment:
should it only acquire it if it is not set? I feel like there's 2 cases here, one if it's not set, and one if it's already defined, i.e., if we set `session[serverSessionSymbol] = 'some random value'`, then we can expect `session.serverSession` to equal that random value

Reply:
tests changed cus of throw